apps/govquests-frontend/src/domains/notifications/components/NotificationToaster.tsx

```diff
@@ -0,0 +1,45 @@
+import { Toaster } from "@/components/ui/toaster";
+import { useToast } from "@/hooks/use-toast";
+import { useEffect } from "react";
+import { useNotificationDiff } from "../hooks/useNotificationDiff";
+import { useNotifications } from "../hooks/useNotifications";
+import { NotificationStrategyFactory } from "../strategies/NotificationStrategyFactory";
+import { NotificationNode } from "../types/notificationTypes";
+
+const NotificationToaster = () => {
+  const { data } = useNotifications();
+  const { toast } = useToast();
+
+  const notifications = data?.pages?.flatMap(
+    (value) => value.notifications.edges,
+  );
+  const { getNewNotifications } = useNotificationDiff(notifications);
+
+  useEffect(() => {
+    const newNotifications = getNewNotifications();
+
+    newNotifications.forEach((notification) => {
+      displayNotification(notification.node, toast);
+    });
+  }, [notifications]);
+
+  return <Toaster />;
+};
+
+export default NotificationToaster;
+
+const displayNotification = (
+  notification: NotificationNode,
+  toast: ReturnType<typeof useToast>["toast"],
+) => {
+  const notificationStrategy = NotificationStrategyFactory.createStrategy(
+    notification.notificationType,
+  );
```

Comment:
this function is not adding much, is it?

Reply:
It isn't! I will call this directly on `NotificationToaster`

Reply:
Actually, `createStrategy` was defined as a static method, should I modify to non static?

Reply:
🤔 why would it be non static?

Reply:
I thought you were saying that I could instantiate the Factory only once before the useEffect. But, since the createStrategy is a static method, it's not necessary to instantiate it. LMK if I understood it incorrectly

Reply:
ah, you're right! good call 👍  feel free to disregard my comment